### PR TITLE
Bug 1877277 - Translations add Check for isTranslationsEngineSupported

### PR DESCRIPTION
**Pull request by iorgamgabriel**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1877277

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -198,7 +198,7 @@ open class DefaultToolbarMenu(
      */
     @VisibleForTesting(otherwise = PRIVATE)
     fun shouldShowTranslations(): Boolean = selectedSession?.let {
```
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:200]: Thanks for adding this check! Can you also block the translations entry point on the [BrowserMenu](https://searchfox.org/mozilla-mobile/rev/c187a25712a5a5bb196c552ed6c14c1df0e3e493/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt#303) too?

File: fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsBinding.kt [hunk 2]
```diff
@@ -68,13 +68,17 @@ class TranslationsBinding(
...
-                if (sessionTranslationsState.isTranslated) {
-                    val fromSelected = sessionTranslationsState.translationEngineState?.initialFromLanguage(
-                        translateFromLanguages,
-                    )
-                    val toSelected = sessionTranslationsState.translationEngineState?.initialToLanguage(
-                        translateToLanguages,
-                    )
+                val isEngineSupported = browserTranslationsState.isEngineSupported == true
+
+                if (sessionTranslationsState.isTranslated && isEngineSupported) {
```
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsBinding.kt:73]: If the `isEngineSupported` on these individual items can be removed, I would recommend removing them to simplify the logic. 

(I think in this scenario, the values just go to null and/or don't change.)

I tested entering the browser menu with `browser.translations.simulateUnsupportedEngine` = true in `about:settings`. This logic nulled it out as expected. Might be worth testing if the `isEngineSupported` are really required.

File: fenix/app/src/test/java/org/mozilla/fenix/browser/TranslationsBindingTest.kt [hunk 3]
```diff
@@ -149,6 +159,11 @@ class TranslationsBindingTest {
                     selectedTabId = tabId,
                 ),
             )
+            browserStore.dispatch(
```
Comment (ohall-m) [inline fenix/app/src/test/java/org/mozilla/fenix/browser/TranslationsBindingTest.kt:162]: If you can think of a test for the browser menu entry point, that would be nice to have too.

**Review by ohall-m — APPROVED** [review]
Thanks for working on this flow! This patch tested very well! I only have one change request, can you also block the entry point on the browser menu too?

I don't want to block for this change, so please additionally manually test after the change by going to `about:config` and set `browser.translations.simulateUnsupportedEngine` to true to manually test that no entry points are showing and the icons layout out as expected on the browser menu.

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt (hunk 1 above)
Comment (iorgamgabriel) [reply fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:200]: Done

**Pull request merged**